### PR TITLE
opusfile: bump openssl req

### PR DESCRIPTION
**Pull request by ericLemanissier**
to avoid conflict with pulseaudio

Specify library name and version:  **opusfile/***

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by SpaceIm** [comment]
Could add `required_conan_version = ">=1.33.0"` please? You can also take advantage of this PR to bump msys2 and move ConanInvalidConfiguration to validate().

**Review by SpaceIm — APPROVED** [review]

**Review by madebr — DISMISSED** [review]

**Comment by ericLemanissier** [comment]
@madebr do you have an idea about this macos error ?
```
Makefile.am:10: error: Libtool library used but 'LIBTOOL' is undefined
Makefile.am:10:   The usual way to define 'LIBTOOL' is to add 'LT_INIT'
Makefile.am:10:   to 'configure.ac' and run 'aclocal' and 'autoconf' again.
Makefile.am:10:   If 'LT_INIT' is in 'configure.ac', make sure
Makefile.am:10:   its definition is in aclocal's search path.
```
should I try to build_require libtool ?

**Comment by madebr** [comment]
> @madebr do you have an idea about this macos error ?
> 
> ```
> Makefile.am:10: error: Libtool library used but 'LIBTOOL' is undefined
> Makefile.am:10:   The usual way to define 'LIBTOOL' is to add 'LT_INIT'
> Makefile.am:10:   to 'configure.ac' and run 'aclocal' and 'autoconf' again.
> Makefile.am:10:   If 'LT_INIT' is in 'configure.ac', make sure
> Makefile.am:10:   its definition is in aclocal's search path.
> ```
> 
> should I try to build_require libtool ?

Yes, `LT_INIT` is mentioned in its `configure.ac`: https://github.com/xiph/opusfile/blob/56728646d626791b08d2b15af0147fb9d0d27bf7/configure.ac#L24

File: recipes/opusfile/all/conanfile.py [hunk 1]
```diff
@@ -41,27 +42,32 @@ def config_options(self):
...
         if self.options.http:
-            self.requires("openssl/1.1.1i")
+            self.requires("openssl/1.1.1k")
 
     def build_requirements(self):
         if not self._is_msvc:
             # FIXME: needs libtool for `autoreconf`, but the `configure.ac` file uses `m4_esyscmd` with bash code as argument.
             # Looks like this does not work with a MSVC-built m4.
             # self.build_requires("libtool/2.4.6")
             if tools.os_info.is_windows and not tools.get_env("CONAN_BASH_PATH"):
```
Comment (SpaceIm) [inline recipes/opusfile/all/conanfile.py:63]:
```suggestion
        if not self._is_msvc:
            self.build_requires("libtool/2.4.6")
            self.build_requires("pkgconf/1.7.4")
            if tools.os_info.is_windows and not tools.get_env("CONAN_BASH_PATH"):
```

File: recipes/opusfile/all/conanfile.py [hunk 2]
```diff
@@ -41,27 +42,32 @@ def config_options(self):
...
-                self.build_requires("msys2/20200517")
+                self.build_requires("msys2/cci.latest")
+        if self.settings.os in ["Linux", "Macos", "FreeBSD"]:
+            self.build_requires("libtool/2.4.6")
+            self.build_requires("pkgconf/1.7.4")
```
Comment (SpaceIm) [inline recipes/opusfile/all/conanfile.py:67]:
```suggestion
```

**Review by SpaceIm — COMMENTED** [review]
You need libtool for autotools build, whatever the OS.
I don't know if pkgconf is required here, but I kept it.

**Comment by SpaceIm** [comment]
By the way, instead of calling autogen.sh, I would call autoreconf manually in the recipe like we do in all others recipe (`self.run("{} -fiv".format(tools.get_env("AUTORECONF")), win_bash=tools.os_info.is_windows)`), to be sure to use autoreconf from autoconf recipe.

**Review by SpaceIm — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Pull request merged**